### ISSUE TITLE
community: Enhancing Community Involvement

### ISSUE THREAD

**Issue by naveensrinivasan** [guacsec/guac#379]
I noticed that the project has "Maintainer meetings (weekly)," which are currently private and not open to the public. Open and transparent communication is crucial for the success of any open-source project and makes it a welcoming environment for all contributors.

I suggest that all meetings are made public and notes or a summary of the discussions be shared publicly so that everyone can stay informed about the project's progress and any important decisions made during the meetings. It's worth mentioning that the agenda can be only among the maintainers and wouldn't take any topic from non-maintainers.

**Comment by naveensrinivasan** [comment]
@mlieberman85 @lumjjb @pxp928 @mihaimaruseac 👀

**Comment by mlieberman85** [comment]
It is definitely something we can discuss. We have plans to put GUAC eventually under more standard governance. We have currently been laser focused on a core set of features.

**Comment by naveensrinivasan** [comment]
> It is definitely something we can discuss. We have plans to put GUAC eventually under more standard governance. We have currently been laser focused on a core set of features.

Do you have specific timelines on these? When would you like to discuss?

**Comment by lumjjb** [comment]
Do we want more community engagement? Yes! :D.  Can we do a better job of it? Probably!

So tl;dr is that we appreciate the feedback. This is evolving as we are getting more contributors and as the project matures, and I can share as much as possible about the thought process as well as what we are planning. We understand what works for others may not work for us and vice versa and it is in constant iteration.

## A bit of what we're trying for community

**Foundation for meaningful contributions:** We would very much want more folks like yourself and others to contribute to GUAC. That said, we want to make sure that we are able to engage with community members in meaningful ways. Being able to do this successfully means the project has structure and has fairly stable interfaces (and docs) where required (e.g. we don't want someone to be working on  an issue and in a month later they come back the entire module is refactored). This is something that we've been working on (it's pretty challenging for what we want to do!). Today, some interfaces are stronger than others, but we have a ways more to go (probably about 3 more months). This is a bit of a side effect of open sourcing a pretty complex project from day 1, and may make things a bit more challenging for early contributors in the project!

Therefore we've taken a split approach with this. For stable interfaces, we've created a couple "good first issues" which we've had great contributions for. 

However, there are certain issues that are very large and complex around some interfaces that need to be tackled. For these, we've tried to break them down into smaller problems - the [identifier/trust problem](https://github.com/guacsec/guac/issues/217), [graphQL refactoring](https://github.com/guacsec/guac/issues/216), [data collect subscription](https://github.com/guacsec/guac/issues/244), [ontology](https://github.com/guacsec/guac/issues/291). Each of them having their own issue and open discussion. For each, usually a contributor or maintainer writes up a design doc, and they post it in the issue for comment. In addition, we want to provide an open forum for the community to discuss on the slack channel and also via community calls ([last one in December](https://drive.google.com/file/d/1u1O6RSYeZT2w6u9jxeSj9X9Z1uqtD1Vn/view?usp=sharing), next one in Feb). The main intention is to have feedback on the direction and to provide suggestions.

We estimate that in about 3 months that these will be pretty solid and we will have minimal foundation changes after. The interfaces we will have will be the core of what will be the 1.0.

>I noticed that the project has "Maintainer meetings (weekly)," which are currently private and not open to the public. 

The weekly meetings are mainly to check in on triage of issues, and making sure we address community feedback or highlight issues that come up, mostly of the nature of "hey we need this PR reviewed, we need 1 more maintainer to look at this". 

On a very rare occasion, we do get together in person to have a more involved discussion. We name these as summits and notes to the summits are released to public. The first one was done prior to creation of the project ([notes from the summit translated to design doc](https://docs.google.com/document/d/1N5x0HErb-kmCPgG9M8TwBEOGIVU54clqp_X4KhtNJI8/edit#heading=h.218ooltp2z7c)). We just had another one last week where we discussed the list of design docs there are public, I'm currently cleaning up the meeting notes to make them more easy to read for folks not as familiar with the design docs we are talking about (adding in context around discussions).


## What's coming up!

- We want to have more community meetings, initially we talked about doing them ad-hoc whenever we felt that there was enough progress and technical issues to talk about. Going forward, we will be doing it monthly. This is going to be scheduled on 16th Feb and recurring on the monthly Thursday.
- The notes from the summit should be cleaned up and released by end of week. We want to make sure that we make the materials as consumable as possible. 
- Once GUAC beta is done (probably in 3 months), we will free up some heads down dev time to community. By then the core structure will be solid for contributions.

## Some ideas that we can riff on

Given the number of code contributions that have increased lately, it may be worthwhile thinking about having higher bandwidth communication for contributors, so maybe having an office hours of some sort - separate from the community meetings, these will focus on technical details. Although some folks may not be fully comfortable with doing this publicly, I think most of us are okay with folks messaging us on slack directly.

**Comment by naveensrinivasan** [comment]
Thanks! I am going to keep this open so that new contributors are aware of the plans.

**Comment by pxp928** [comment]
Added [CONTRIBUTING.md](https://github.com/guacsec/guac/blob/main/CONTRIBUTING.md) with Contributor ladder. @naveensrinivasan @lumjjb close this issue?

**Comment by lumjjb** [comment]
yea let's close this now, it looks like the comments have been addressed with the new docs. Feedback on the docs should be a new issue.
